julia/mmtk_julia.c

```diff
@@ -91,41 +95,42 @@ JL_DLLEXPORT jl_value_t *jl_mmtk_gc_alloc_big(jl_ptls_t ptls, size_t sz)
     }
     v->sz = allocsz;
 
-    ptls->gc_num.allocd += allocsz;
-    ptls->gc_num.bigalloc++;
+    ptls->gc_tls.gc_num.allocd += allocsz;
+    ptls->gc_tls.gc_num.bigalloc++;
 
     jl_value_t *result = jl_valueof(&v->header);
     mmtk_post_alloc(&ptls->mmtk_mutator, result, allocsz, 2);
 
     return result;
 }
 
-static void mmtk_sweep_malloced_arrays(void) JL_NOTSAFEPOINT
+static void mmtk_sweep_malloced_memory(void) JL_NOTSAFEPOINT
```

Comment:
There are very few differences between this function and the stock one. Maybe this should live (if possible) in the `mmtk-gc.c` file instead of here and be part of the `gc.h` api?

Reply:
I think eventually this should go to `gc-common` or something. It takes a function pointer to check if an object is alive or not. The rest of the code can be reused for both the stock GC and MMTk. We don't need to do it now. I am just saying with the interface change, we should start sorting out this kind of duplicated code.